source/menu/events/add.ts

```diff
@@ -1,42 +1,36 @@
 import {StatelessQuestion} from '@grammyjs/stateless-question';
 import {Composer} from 'grammy';
 import {
-	deleteMenuFromContext,
-	getMenuOfPath,
-	MenuTemplate,
-	replyMenuToContext,
+	deleteMenuFromContext, getMenuOfPath, MenuTemplate, replyMenuToContext,
 } from 'grammy-inline-menu';
 import {html as format} from 'telegram-format';
 import {
-	count as allEventsCount,
-	exists as allEventsExists,
-	find as allEventsFind,
+	count as allEventsCount, directoryExists, find as allEventsFind, getEventName,
 } from '../../lib/all-events.ts';
-import {
-	DEFAULT_FILTER,
-	filterButtonText,
-} from '../../lib/inline-menu-filter.ts';
-import {backMainButtons} from '../../lib/inline-menu.ts';
-import type {MyContext} from '../../lib/types.ts';
+import {filterButtonText} from '../../lib/inline-menu-filter.ts';
+import {BACK_BUTTON_TEXT} from '../../lib/inline-menu.ts';
+import type {EventDirectory, EventId, MyContext} from '../../lib/types.ts';
+import {getUserEventIdsFromContext} from '../../lib/calendar-helper.js';
 
 const MAX_RESULT_ROWS = 10;
-const RESULT_COLUMNS = 2;
+const RESULT_COLUMNS = 1;
 
 export const bot = new Composer<MyContext>();
 export const menu = new MenuTemplate<MyContext>(async ctx => {
-	const total = await allEventsCount();
+	const total = allEventsCount();
+	ctx.session.eventPath ??= [];
 
 	let text = format.bold('Veranstaltungen');
 	text += '\nWelche Events möchtest du hinzufügen?';
 	text += '\n\n';
 
 	try {
-		const filteredEvents = await findEvents(ctx);
+		const filteredEvents = findEvents(ctx);
 
-		const filter = ctx.session.eventfilter ?? DEFAULT_FILTER;
-		text += filter === DEFAULT_FILTER
+		const filter = ctx.session.eventfilter;
+		text += filter === undefined
 			? `Ich habe ${total} Veranstaltungen. Nutze den Filter um die Auswahl einzugrenzen.`
-			: `Mit deinem Filter konnte ich ${filteredEvents.length} passende Veranstaltungen finden.`;
+			: `Mit deinem Filter konnte ich ${Object.keys(filteredEvents.events).length} passende Veranstaltungen und ${Object.keys(filteredEvents.subDirectories).length} Ordner finden.`;
```

Comment:
Vermutlich wirds leichter lesbar, wenn die `Object.keys().length` Dinger in Variablen enden und nur noch variablen im String template sind. Vielleicht auch gleich mit richtigem if/else statt ternary wie bisher.